lib/galaxy/model/security.py

```diff
@@ -1447,62 +1439,171 @@ def get_showable_folders(
                 self.get_showable_folders(user, roles, folder, actions_to_check, showable_folders=showable_folders)
         return showable_folders
 
-    def set_entity_user_associations(self, users=None, roles=None, groups=None, delete_existing_assocs=True):
-        users = users or []
-        roles = roles or []
-        groups = groups or []
-        for user in users:
-            if delete_existing_assocs:
-                flush_needed = False
-                for a in user.non_private_roles + user.groups:
-                    self.sa_session.delete(a)
-                    flush_needed = True
-                if flush_needed:
-                    with transaction(self.sa_session):
-                        self.sa_session.commit()
-            self.sa_session.refresh(user)
-            for role in roles:
-                # Make sure we are not creating an additional association with a PRIVATE role
-                if role not in [x.role for x in user.roles]:
-                    self.associate_components(user=user, role=role)
-            for group in groups:
-                self.associate_components(user=user, group=group)
+    def set_user_group_and_role_associations(
+        self,
+        user: User,
+        *,
+        group_ids: Optional[List[int]] = None,
+        role_ids: Optional[List[int]] = None,
+    ) -> None:
+        """
+        Set user groups and user roles, replacing current associations.
 
-    def set_entity_group_associations(self, groups=None, users=None, roles=None, delete_existing_assocs=True):
-        users = users or []
-        roles = roles or []
-        groups = groups or []
-        for group in groups:
-            if delete_existing_assocs:
-                flush_needed = False
-                for a in group.roles + group.users:
-                    self.sa_session.delete(a)
-                    flush_needed = True
-                if flush_needed:
-                    with transaction(self.sa_session):
-                        self.sa_session.commit()
-            for role in roles:
-                self.associate_components(group=group, role=role)
-            for user in users:
-                self.associate_components(group=group, user=user)
+        Associations are set only if a list of new associations is provided.
+        If the provided list is empty, existing associations will be removed.
+        If the provided value is None, existing associations will not be updated.
+        """
+        self._ensure_model_instance_has_id(user)
+        if group_ids is not None:
+            self._set_user_groups(user, group_ids or [])
```

Comment:
```suggestion
            self._set_user_groups(user, group_ids)
```
I assume if it's not None it must be a list ?

Reply:
Yes, that was left over from a previous iteration. Thanks!